numpy/random/tests/test_regression.py

```diff
@@ -113,5 +113,20 @@ def test_choice_sum_of_probs_tolerance(self):
             assert_(c in a)
             assert_raises(ValueError, np.random.choice, a, p=probs*0.9)
 
+    def test_permutation_of_array_of_different_length_strings(self):
+        # Test that permuting an array of different length strings
+        # will not cause a segfault on garbage collection
+        # Tests gh-7710
+        np.random.seed(1234)
+
+        a = np.array(['a', 'a' * 1000])
+
+        for _ in range(100):
+            z = np.random.permutation(a)
+
+        # Force Garbage Collection - should not segfault.
```

Comment:
The explicit garbage collection is to trigger the segfault within the test rather than at some arbitrary time in the future (it could even be when the interpreter exits). Thus if this test fails sometime in the future we will know what the failure is!